enterprise/backend/src/metabase_enterprise/metabot_v3/table_utils.clj

```diff
@@ -153,3 +158,61 @@
         {recognized-tables true, unrecognized-tables false} (group-by t2/instance? queried-tables)]
     (concat recognized-tables
             (find-matching-tables database unrecognized-tables (map :id recognized-tables)))))
+
+(defn- format-escaped
+  [^String identifier ^Writer writer]
+  (if (re-matches #"[\p{Alpha}_][\p{Alnum}_]*" identifier)
+    (.append writer identifier)
+    (do
+      (.append writer \")
+      (doseq [^Character c identifier]
+        (when (= c \")
+          (.append writer c))
+        (.append writer c))
+      (.append writer \"))))
+
+(defn- format-field-ddl
+  [{:keys [^String database_type], fname :name} ^Writer writer]
+  (format-escaped fname writer)
+  (.append writer " ")
+  (.append writer database_type))
+
+(defn- format-table-ddl
+  [{:keys [schema fields], tname :name} ^Writer writer]
+  (.append writer "CREATE TABLE ")
+  (when (seq schema)
+    (format-escaped schema writer)
+    (.append writer \.))
+  (format-escaped tname writer)
+  (.append writer " (")
+  (when (seq fields)
+    (.append writer "\n  ")
+    (format-field-ddl (first fields) writer)
+    (doseq [field (rest fields)]
+      (.append writer ",\n  ")
+      (format-field-ddl field writer)))
+  (.append writer "\n);"))
+
+(defn- format-schema-ddl
+  [tables]
+  (let [sw (StringWriter.)]
+    (doseq [table tables]
+      (format-table-ddl table sw)
+      (.append sw \newline))
+    (str sw)))
+
+(defn schema-sample
```

Comment:
Tests for this? I can also add them afterwards.

Reply:
Also do we have something already does this elsewhere?

Reply:
I moved this here from the private function it used to be